framework/auth/views.py

```diff
@@ -825,3 +820,27 @@ def external_login_email_post():
         'form': form,
         'external_id_provider': external_id_provider
     }
+
+
+def validate_next_url(next_url):
```

Comment:
For Reviewer: this is used in the another PR https://github.com/CenterForOpenScience/osf.io/pull/6082 (CAS Institution Login): https://github.com/CenterForOpenScience/osf.io/pull/6082/files#diff-9a64b4982d0bd80f7a498c65ddce2023R283.